### PR TITLE
feat: AVS reference

### PR DESCRIPTION
**Pull request by aroralanuk**
Added the reference guide for the AVS.

### CODE REVIEW

File: docs/protocol/avs.mdx [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+
+Staker->>DelegationManager: delegateTo(operator,approverSignatureAndExpiry)
+Staker->>StrategyManager: depositIntoStrategy(strategy,token,amt)
+StrategyManager->>Strategy1: token.transferFrom(sender, strategy, amount)
+StrategyManager->>DelegationManager: increaseDelegatedShares
+```
+
+In Eigenlayer's design, staker funds are safeguarded as they are not made accessible to the AVS until slashing is activated. This ensures that no funds are at risk prematurely. Additionally, this design does not allow users to selectively opt into the Hyperlane AVS alone, as staking is managed through the StrategyManager and directed towards a specific operator.
+
+### Workflow for reading stake from an ISM
```
Comment (nambrot) [inline docs/protocol/avs.mdx:172]: I would either remove this section for now, or move it to the bottom or a subpage

File: docs/protocol/avs.mdx [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+
+Both the challenger and slasher contracts are not yet live in production. This sequence diagram can change based on the final implementation.
+
+:::
+
+### Protocol SLA
+
+Anyone can join Hyperlane as a validator, i.e., there's no defacto validator set. You can start attesting to messages from your chain but you'll only be rewarded if you are picked by an [ISM](./ISM/modular-security.mdx) deployer for verification. Similarily, you can only get slashed by the said ISM deployer if you enroll into their Challenger.
+
+Because of the permissionless nature of the protocol, Hyperlane wouldn't be slashing for liveness but you can still be penalized by not being selected in an ISM's validator set and losing out on rewards.
```
Comment (nambrot) [inline docs/protocol/avs.mdx:235]: There is no rewards mechanism setup yet, so i would remove that.

File: docs/protocol/avs.mdx [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+
+:::note
+
+Both the challenger and slasher contracts are not yet live in production. This sequence diagram can change based on the final implementation.
+
+:::
+
+### Protocol SLA
+
+Anyone can join Hyperlane as a validator, i.e., there's no defacto validator set. You can start attesting to messages from your chain but you'll only be rewarded if you are picked by an [ISM](./ISM/modular-security.mdx) deployer for verification. Similarily, you can only get slashed by the said ISM deployer if you enroll into their Challenger.
```
Comment (nambrot) [inline docs/protocol/avs.mdx:233]: Sometimes I would read this as a 1:1 correspondence between ISM deployer and challenger. I dont expect that to be true, so I would be consistent and just say that you will be slashed by the challenger.

Comment (nambrot) [reply docs/protocol/avs.mdx:233]: Ditto re: rewards, would just omit ISM selection. At the top I would just be explicit that enrollment and operation does not mean that the signature actually secure messages until setup in an ISM

File: sidebars.js [hunk 4]
```diff
@@ -272,6 +272,11 @@ const sidebars = {
       id: "protocol/protocol-overview",
       label: "Overview",
     },
+    {
```
Comment (nambrot) [inline sidebars.js:275]: But this below the ISM section, not above Mailbox on the sidebar

**Review by nambrot — APPROVED** [review]
I'm fine with having the protocol docs comments be addressed in another PR to get it updated asap

File: docs/guides/avs-operator-guide.mdx [hunk 5]
```diff
@@ -4,27 +4,10 @@ Hyperlane validators are extremely lightweight off-chain agents responsible for
...
-:::
-
-To register with Hyperlane, operators must first register as an operator on the Eigenlayer smart contracts on Holesky testnet. You can follow Eigenlayer's instructions [here](https://docs.eigenlayer.xyz/eigenlayer/operator-guides/operator-installation#operator-configuration-and-registration) (only need the ECDSA key).
-
-Next, to register your operator to Hyperlane AVS, you can register your operator key (generated via EigenLayer CLI) by creating an ECDSA signature and submitting it via the Hyperlane CLI:
-
-```bash
- hyperlane avs register --operatorKeyPath <PATH_TO_OPERATOR_KEY> --chain holesky --key <HYP_KEY>
-```
+- Install the latest version of the `@hyperlane-xyz/cli` package with `npm install -g @hyperlane-xyz/cli`. Follow more detailed instructions for installation [here](https://docs.hyperlane.xyz/docs/reference/cli).
```
[the remark this answers is not in the document]
Comment (aroralanuk) [reply docs/guides/avs-operator-guide.mdx:7]: Yes

File: docs/protocol/avs.mdx (hunk 2 above)
Comment (aroralanuk) [reply docs/protocol/avs.mdx:235]: omitted the SLA section.

**Pull request merged**